File: 1440b/main.py
def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))

# alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
a = II()
for _ in range(a):
    n, k = MII()
    L = LII()

    x = n // 2
    ans = 0
    tmpk = 0

    while L:
        tmp = x
        while tmp:
            L.pop()
            tmp -=1
        ans += L.pop()
        tmpk += 1
        if tmpk == k:
            break
    print(ans)

File: 1440b/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func solve(reader *bufio.Reader) {
	var n, k int
	fmt.Fscan(reader, &n, &k)
	L := make([]int, n*k)
	for i := 0; i < n*k; i++ {
		fmt.Fscan(reader, &L[i])
	}
	ans := 0
	tmpk := 0
	p := n*k - 1
	x := n / 2

	for {
		tmp := x
		//println(tmp)
		for tmp > 0 {
			p -= 1
			tmp -= 1
		}

		//println(p, L[p])
		ans += L[p]
		p -= 1
		tmpk += 1
		if tmpk == k {
			break
		}
	}
	fmt.Println(ans)
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	var a int
	fmt.Fscan(reader, &a)
	for i := 0; i < a; i++ {
		solve(reader)
	}
}
